crates/router/src/core/payouts.rs

```diff
@@ -443,11 +646,31 @@ pub async fn call_connector_payout(
     merchant_account: &domain::MerchantAccount,
     key_store: &domain::MerchantKeyStore,
     req: &payouts::PayoutCreateRequest,
-    connector_data: api::PayoutConnectorData,
+    connector_data: &api::ConnectorData,
     payout_data: &mut PayoutData,
-) -> RouterResponse<payouts::PayoutCreateResponse> {
+) -> RouterResult<PayoutData> {
     let payout_attempt = &payout_data.payout_attempt.to_owned();
     let payouts: &diesel_models::payouts::Payouts = &payout_data.payouts.to_owned();
+
+    // update connector_name
+    payout_data.payout_attempt.connector = Some(connector_data.connector_name.to_string());
```

Comment:
Can we update the connector in the current state and DB only when it is re-evaluated?

Calling this here will make a DB query every time a connector is to be called for creation / updation.